### ISSUE TITLE
[API Compat] Add rule to prohibit making a member virtual

### ISSUE THREAD

**Issue by safern** [dotnet/sdk#18683]
Adding the virtual keyword to a member is a breaking change because the consumer code would often end up calling it non-virtually when the member is now virtual.

This rule should just run when not in strict mode as when running on strict mode that will be covered by: https://github.com/dotnet/sdk/issues/18682

cc: @ericstj @Anipik

**Comment by smasher164** [comment]
Make sure to update docs repo after #25734 goes in.

**Comment by ericstj** [comment]
As discussed this was not enforced by the previous API compat, and we might not consider this actually breaking since code compiled against the old non-virtual method continues to work and call the same method.

**Comment by smasher164** [comment]
> @smasher164 should this issue be closed now?

Right now, this rule is only enabled in strict mode, since *adding* a virtual keyword is used in a ton of places while intending to preserve backwards compatibility. There is a `TODO` comment in the rule that explains that we can expand it past strict mode based on future discussion: https://github.com/dotnet/sdk/blob/main/src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Rules/CannotAddOrRemoveVirtualKeyword.cs#L67.

I think we should close this issue, since we have already the TODO comment that links to a separate issue.
